source/backend/api/Helpers/Middleware/ErrorHandlingMiddleware.cs

```diff
@@ -130,6 +130,13 @@ private async Task HandleExceptionAsync(HttpContext context, Exception ex)
 
                 _logger.LogWarning(ex, ex.Message);
             }
+            else if (ex is MayanRepositoryException)
```

Comment:
Am I missing something? I see that you've added a new MayanRepositoryException class, but I don't see that being thrown anywhere.